env.go

```diff
@@ -125,6 +127,7 @@ func (e *Environment) Go(f func(ctx context.Context) error) {
 	go func() {
 		ctx, cancel := context.WithCancel(e.ctx)
 		defer cancel()
+		ctx = WithRequestID(ctx, e.generator.Generate())
```

Comment:
If Go is called in ServeHTTP, its request ID is overwritten.
This is possibly undesired behavior, but invoking HTTPClient.Do also does not inherit context automatically.
So, we conclude the behavior is sane in an internal discussion.
(If we have better idea, this behavior would be changed.)